app/controllers/backoffice/data_layers_controller.rb

```diff
@@ -18,4 +18,8 @@ def data_layer_params
     params.permit(:id)
   end
 
+  def destroy
+    layer = DataLayer.where(id:data_layer_params[:id]).first
```

Comment:
If you are looking for just one object no need to use where and first. just use `DataLayer.find(params[:id])`

And the method `data_layer_params` should really be after the all the actions in a `private` block.  See: https://github.com/Vizzuality/esaworld/blob/master/app/controllers/backoffice/pages_controller.rb#L59

Also the current Rails trend is to create a `set_data_layer` method and add a `before_action` at the top.  Check the same controller pasted above: https://github.com/Vizzuality/esaworld/blob/master/app/controllers/backoffice/pages_controller.rb#L4